File: app/src/main/java/com/elementary/tasks/navigation/fragments/FeedbackFragment.java
package com.elementary.tasks.navigation.fragments;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.elementary.tasks.R;
import com.elementary.tasks.core.utils.Module;

/**
 * Copyright 2016 Nazar Suhovich
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

public class FeedbackFragment extends BaseWebViewFragment {

    @Override
    protected String getUrl() {
        return "https://docs.google.com/forms/d/1vOCBU-izJBQ8VAsA1zYtfHFxe9Q1-Qm9rp_pYG13B1s/viewform";
    }

    @Override
    protected void setExtraParams(WebView webView) {
        super.setExtraParams(webView);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient() {
            public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
            }

            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                if (url != null && url.contains("https://bitbucket.org/nazar_suhovich/just-reminder/issues?status=new&status=open")) {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(url)));
                    return true;
                } else {
                    return false;
                }
            }
        });
        webView.setWebChromeClient(new WebChromeClient());
    }

    @Override
    public void onResume() {
        super.onResume();
        if (getCallback() != null) {
            getCallback().onTitleChange(getString(R.string.feedback));
            getCallback().onFragmentSelect(this);
        }
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.menu_feedback, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_refresh:
                getWebView().reload();
                return true;
            case R.id.action_forward:
                if (getWebView().canGoForward()) {
                    getWebView().goForward();
                }
                return true;
            case R.id.action_back:
                if (getWebView().canGoBack()) {
                    getWebView().goBack();
                }
                return true;
            case R.id.action_email:
                sendEmail();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void sendEmail() {
        final Intent emailIntent = new Intent(Intent.ACTION_SEND);
        emailIntent.setType("plain/text");
        emailIntent.putExtra(Intent.EXTRA_EMAIL, new String[]{"feedback.cray@gmail.com"});
        if (Module.isPro()) {
            emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Reminder PRO");
        } else {
            emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Reminder");
        }
        startActivity(Intent.createChooser(emailIntent, "Send mail..."));
    }
}

File: app/src/main/java/com/elementary/tasks/navigation/fragments/FeedbackFragment.kt
package com.elementary.tasks.navigation.fragments

import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import com.elementary.tasks.R
import com.elementary.tasks.core.utils.Module
import com.elementary.tasks.core.utils.ui.ViewUtils

class FeedbackFragment : BaseWebViewFragment() {

  override val url: String
    get() = FORM_URL

  @SuppressLint("SetJavaScriptEnabled")
  override fun setExtraParams(webView: WebView) {
    super.setExtraParams(webView)
    webView.settings.javaScriptEnabled = true
    webView.webViewClient = object : WebViewClient() {
      @Deprecated("Deprecated in Java")
      override fun onReceivedError(
        view: WebView,
        errorCode: Int,
        description: String,
        failingUrl: String
      ) {
      }

      @Deprecated("Deprecated in Java")
      override fun shouldOverrideUrlLoading(view: WebView, url: String?): Boolean {
        return if (url != null && url.contains(GITHUB_URL)) {
          startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
          true
        } else {
          false
        }
      }
    }
    webView.webChromeClient = WebChromeClient()
  }

  override fun getTitle(): String = getString(R.string.feedback)

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    addMenu(R.menu.fragment_feedback, { menuItem ->
      return@addMenu when (menuItem.itemId) {
        R.id.action_refresh -> {
          webView.reload()
          true
        }

        R.id.action_forward -> {
          if (webView.canGoForward()) {
            webView.goForward()
          }
          true
        }

        R.id.action_back -> {
          if (webView.canGoBack()) {
            webView.goBack()
          }
          true
        }

        R.id.action_email -> {
          sendEmail()
          true
        }

        else -> false
      }
    }) {
      ViewUtils.tintMenuIcon(
        context = requireContext(),
        menu = it,
        index = 0,
        resource = R.drawable.ic_fluent_arrow_clockwise,
        isDark = isDark
      )
      ViewUtils.tintMenuIcon(
        context = requireContext(),
        menu = it,
        index = 1,
        resource = R.drawable.ic_builder_send_message,
        isDark = isDark
      )
    }
    invalidateOptionsMenu()
  }

  private fun sendEmail() {
    val emailIntent = Intent(Intent.ACTION_SEND)
    emailIntent.type = "plain/text"
    emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("feedback.cray@gmail.com"))
    if (Module.isPro) {
      emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Reminder PRO")
    } else {
      emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Reminder")
    }
    startActivity(Intent.createChooser(emailIntent, "Send mail..."))
  }

  companion object {
    const val FORM_URL = "https://docs.google.com/forms/d/e/1FAIpQLSdn_x9tnVgG9CBmA30vHmGhDhtS-" +
      "gHoq3RDnuNhJPzr2y15LQ/viewform?usp=sf_link"
    const val GITHUB_URL = "https://github.com/naz013/reminder-kotlin/issues"
  }
}
